packages/esm-patient-search-app/src/patient-search-page/refine-search/person-attribute-field.component.tsx

```diff
@@ -0,0 +1,232 @@
+  const { t } = useTranslation();
+  const { data: personAttributeType, isLoading, error } = usePersonAttributeType(field.attributeTypeUuid);
+
+  const formatField = useMemo(() => {
+    if (!personAttributeType || isLoading) {
+      return <TextInputSkeleton />;
+    }
+
+    switch (personAttributeType.format) {
+      case 'java.lang.String':
+        return (
+          <TextInput
+            id={field.name}
+            labelText={t(field.label || personAttributeType.display)}
+            value={formState.attributes[field.name] || ''}
+            onChange={(e: { target: { value: any } }) => {
+              onInputChange(field.name)({ target: { value: e.target.value } });
+            }}
+            placeholder={field.placeholder}
+            size={isTablet ? 'lg' : 'md'}
+          />
+        );
+
+      case 'org.openmrs.Concept':
+        return (
+          <ConceptAttributeField
+            field={field}
+            formState={formState}
+            isTablet={isTablet}
+            onInputChange={onInputChange}
+            attributeDisplay={personAttributeType.display}
+          />
+        );
+
+      case 'org.openmrs.Location':
+        return (
+          <LocationAttributeField
+            field={field}
+            formState={formState}
+            isTablet={isTablet}
+            onInputChange={onInputChange}
+            attributeDisplay={personAttributeType.display}
+          />
+        );
+
+      default:
+        return (
+          <InlineNotification kind="error" title={t('error', 'Error')}>
+            {t('unsupportedAttributeFormat', 'Unsupported attribute format: {{format}}', {
+              format: personAttributeType.format,
+            })}
+          </InlineNotification>
+        );
+    }
+  }, [personAttributeType, isLoading, field, formState, onInputChange, t, isTablet]);
+
+  if (error) {
+    return (
+      <InlineNotification kind="error" title={t('error', 'Error')}>
+        {t('errorLoadingAttribute', 'Error loading attribute type')}
+      </InlineNotification>
+    );
+  }
+
+  return formatField;
+}
+
+const ConceptAttributeField = ({
+  field,
+  formState,
+  isTablet,
+  onInputChange,
+  attributeDisplay,
+}: {
+  field: SearchFieldConfig;
+  formState: AdvancedPatientSearchState;
+  isTablet: boolean;
+  onInputChange: (fieldName: string) => (evt: { target: { value: string } } | { name: string }) => void;
+  attributeDisplay: string;
+}) => {
+  const { t } = useTranslation();
+  const {
+    data: conceptAnswers,
+    isLoading,
+    error,
+  } = useAttributeConceptAnswers(field.customConceptAnswers?.length ? '' : field.answerConceptSetUuid);
+  const handleChange = onInputChange(field.name);
+
+  const items = useMemo(() => {
+    if (field.customConceptAnswers?.length) return field.customConceptAnswers;
+    if (!conceptAnswers || isLoading) return [];
+    return conceptAnswers
+      .map((answer) => ({
+        uuid: answer.uuid,
+        label: answer.display,
```

Comment:
Please use `display` instead of label wherever possible, since we use `display` from the backend.